### PR TITLE
accelerators: assign iop shortcuts to preferred instance by default

### LINKED ISSUE #10148: Input NG: Always default to preferred instance
The behaviour of the visual shortcut mapping seems to differ based on whether there are multiple instances of a module.

- If there's only one instance, the shortcut is applied to the "preferred" instance
- If there are multiple instance, the shortcut seems to be applied to the first instance if I point to the first instance, the second if I point to the second, and the "preferred" if I point to the last

IMO this is not desirable behaviour. Shortcuts should always be applied to the preferred instance by default and the user should have to go into the "advanced" shortcuts window to change that. I don't personally see a use case for applying shortcuts to specific numbered instances but I suspect if it is used it's much more rarely needed than the "preferred" one.

Edit: looking at the code it seems to be setting it to the preferred instance if the one I assign it to happens to be the currently-preferred instance, and to a specific instance number otherwise. Either way I'd prefer it not to do this.

### PR DESCRIPTION
**Pull request by elstoc**
Resolves #10148

### CODE REVIEW

**Comment by dterrahe** [comment]
This removes the feature that allows you to immediately assign preferred instances visually and would require you to _always_ go into the dialog to set them for everything. One of my goals of the visual mapping tool is to allow people to quickly (and possibly temporarily) assign for example midi rotors to a whole set of sliders, if they are doing some advanced edits on a set of images using, say, color balance. At the moment you can very quickly create two sets of mappings for all 8 rotors on the x-touch, one for the first instance of a tool and, holding shift for example, another for the second instance. This PR forces you to afterwards go into the dialog and change all of them.

I do however completely understand that this feature may be surprising, even if the toast that appears after a mapping mentions the instance the shortcut was set up for. So my preferred solution would be to put this behind a (hidden) configuration option "accel/assign_instance", off by default, rather than removing it completely as this PR does, so I don't have to lose this functionality myself.

File: src/gui/accelerators.c [hunk 1]
```diff
@@ -2605,42 +2605,6 @@ static void _lookup_mapping_widget()
   if(!_sc.action) return;
 
   _sc.instance = 0;
-  if(_sc.action->target != darktable.control->mapping_widget)
```
Comment (dterrahe) [inline src/gui/accelerators.c:2608]: Please change this to be conditional on "accel/assign_instance"

**Comment by elstoc** [comment]
Ok I can put it behind a new config item in `preferences > miscellaneous > keyboard shortcuts with multiple instances > allow visual assignment to specific instances` with a tooltip that says "when multiple instances are present on an image this allows shortcuts to be visually assigned to those specific instances. otherwise shortcuts will always be assigned to the preferred instance"

Is that ok with everyone?

**Review by dterrahe — CHANGES_REQUESTED** [review]
Please add the configuration variable "accel/assign_instance" and make the instance lookup code conditional on this rather than removing it.

**Comment by dterrahe** [comment]
> Is that ok with everyone?

Absolutely, but I'd also be content if this was a hidden config option. Obviously default is FALSE.

**Comment by elstoc** [comment]
TBH I'm not a fan of hidden options. I'll make it visible for now.

**Comment by dterrahe** [comment]
> TBH I'm not a fan of hidden options.

There are many things you are not a fan of ;-)

**Comment by elstoc** [comment]
Done. I've also slightly tweaked the section title for the "keyboard shortcuts" section so that it just says "shortcuts".

**Review by dterrahe — APPROVED** [review]
Thank you!

**Review by TurboGit — APPROVED** [review]
Thanks !

**Pull request merged**